snap/validate.go

```diff
@@ -366,6 +366,16 @@ func ValidateBase(info *Info) error {
 	if info.Base == "none" && (len(info.Hooks) > 0 || len(info.Apps) > 0) {
 		return fmt.Errorf(`cannot have apps or hooks with base "none"`)
 	}
+
+	if info.Base != "" {
+		baseSnapName, instanceKey := SplitInstanceName(info.Base)
+		if instanceKey != "" {
+			return fmt.Errorf("invalid base %q: snap instance not allowed", info.Base)
+		}
+		if err := ValidateName(baseSnapName); err != nil {
+			return fmt.Errorf("invalid base: %s", err)
```

Comment:
"invalid base name: %q"

Reply:
Done, made it %s though as this is an err (and snap name is quoted in the err).